app/helpers/upload_display_helper.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2022-2022, Puzzle ITC. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+module UploadDisplayHelper
+  # This method provides a facade to serve uploads either from ActiveStorage or
+  # CarrierWave
+  #
+  # Usage:
+  #
+  # upload_url(person, :picture)
+  # upload_url(person, :picture, size: '72x72')
+  # upload_url(person, :picture, size: '72x72')
+  # upload_url(person, :picture, variant: :thumb)
+  # upload_url(person, :picture, variant: :thumb, default: 'profil')
+  #
+  # could be
+  #
+  # person.picture or
+  # person.picture.variant(resize_to_limit: [72, 72]) or
+  # person.picture.variant(:thumb)
+  #
+  # This helper returns a suitable first argument for image_tag (the image location),
+  # but also for the second arg of link_to (the target).
+  def upload_url(model, name, size: nil, default: model.class.name.underscore, variant: nil) # rubocop:disable Metrics/MethodLength,Metrics/PerceivedComplexity
+    return upload_variant(model, name, variant, default: default) if variant.present?
+
+    if model.send(name.to_sym).attached?
+      model.send(name.to_sym).yield_self do |pic|
+        if size
+          # variant passes to mini_magick or vips, I assume mini_magick here
+          pic.variant(resize_to_limit: extract_image_dimensions(size))
+        else
+          pic
+        end
+      end
+    elsif model.respond_to?(:"carrierwave_#{name}") && model.send(:"carrierwave_#{name}")
+      model.send(:"carrierwave_#{name}_url")
```

Comment:
If this ever happens on a production env, I as a dev would want to know about it. Maybe you could send a message to sentry, without actually raising an exception?

Reply:
Hmm. this helper is transitional in nature and not intended to be there after the data-migration is done. We have #1658 to clean this up. I will think a little if adding notifications is easily possible (with detection of wether the migration is done already).